packages/ember-glimmer/lib/utils/references.js

```diff
@@ -52,6 +52,27 @@ export class UpdatableReference extends RootReference {
   }
 }
 
+export class GetHelperReference {
+  constructor(args) {
+    this.positionalArgs = args.positional;
+  }
+
+  isDirty() { return true; }
+
+  value() {
+    let objRef = this.positionalArgs.at(0).value(),
+      pathRef = this.positionalArgs.at(1).value();
+
+    return objRef ? get(objRef, pathRef) : objRef;
+  }
+
+  get(propertyKey) {
+    throw new Error("get() is not implemented");
```

Comment:
Another thing to understand is why you need `PathReference` in the first place. Certainly, you can just call `.value()`  on the parent and use a regular property lookup on the reified value.

The reason for `PathReference` is that the underlying root reference might have additional information about when you need to reify the value for paths derived off of it:

1. in Ember, if nobody called `.set()` on the parent object, you don't need to recompute the value.
2. in Ember, if you're looking at a computed property, you can rely on the computed property cache.
3. when working with immutable data structures, you only need to recompute the value if the root of the data structure has been replaced.
4. when working with primitive values, the `get()` virtually always produces a null value.
5. when working with non-configurable, non-writable values, you only need to compute the value once.

In other words, there's all kinds of information in the JavaScript **data** layer about how to most-efficiently derive a path off of a value, and `PathReference` allows the owner of the data to encode that strategy.

In Ember, this results in significant efficiency gains when Ember objects are used. Those gains would be portable to other data management strategies like immutable.js without forcing every data structure in your entire app to use Ember objects or immutable.js.